src/mono/wasm/runtime/binding_support.js

```diff
@@ -497,7 +497,7 @@ var BindingSupportLib = {
 			// you need to use a view. A view provides a context — that is, a data type, starting offset, 
 			// and number of elements — that turns the data into an actual typed array.
 			// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Typed_arrays
-			if (!!(js_obj.buffer instanceof ArrayBuffer && js_obj.BYTES_PER_ELEMENT)) 
+			if (!!((js_obj.buffer instanceof ArrayBuffer || js_obj.buffer instanceof SharedArrayBuffer) && js_obj.BYTES_PER_ELEMENT))
```

Comment:
Needs more guarding around the type? e.g. will throw in FireFox
```
Uncaught ReferenceError: SharedArrayBuffer is not defined
```

Reply:
👍

Reply:
Please see recent commit that should address this.  Thank you for the heads up.  Totally forgot about the non supported.